server/src/main/java/org/elasticsearch/index/mapper/TextParams.java

```diff
@@ -129,7 +129,33 @@ private NamedAnalyzer wrapAnalyzer(NamedAnalyzer a) {
     }
 
     public static Parameter<String> indexOptions(Function<FieldMapper, String> initializer) {
```

Comment:
I think I'd prefer these to be `keywordIndexOptions` and `textIndexOptions`, as elsewhere `indexOptions` implies the full set of lucene options.

Reply:
++ renamed :)